src/memory/memory.ts

```diff
@@ -59,4 +56,18 @@ export class Memory {
   getSegmentSize(segment: number): number {
     return this.values[segment]?.length ?? 0;
   }
+
+  toString(): string {
+    const memoryToPrint = [
+      '\nMEMORY',
+      'Address  ->  Value',
+      '-----------------',
+    ];
+    for (const [index, segment] of this.values.entries()) {
+      for (const [offset, value] of segment.entries()) {
+        memoryToPrint.push(`${index}:${offset} -> ${value.toString()}`);
+      }
+    }
+    return memoryToPrint.join('\n');
```

Comment:
I may be biased but in the case of a "pure" interation, I think that map is 100x better than for loop to explicitly show that you don't mix values from one interation to another, but just apply the same thing to each item

```
this.values
    .map((segment, index) =>
    segment.map(
        (value, offset) => `${index}:${offset} -> ${value.toString()}`
    )
    )
    .flat()
    .join('\n');
```

Reply:
Indeed, much cleaner